keg_bouncer/model/mixins.py

```diff
@@ -1,21 +1,37 @@
 from __future__ import absolute_import
 
 import sqlalchemy as sa
+from sqlalchemy.inspection import inspect
 from sqlalchemy.ext.declarative import declared_attr
 import sqlalchemy.orm as saorm
 
-import flask_user
+from . import entities as ents
 
-from .entities import (
-    UserGroup,
-    joined_permission_query,
-    make_user_to_user_group_link,
-    user_group_bundle_map,
-    user_group_permission_map,
-)
 
+class KegBouncerMixin(object):
+    _invalid_primary_key_error = AttributeError(
```

Comment:
I wouldn't even inherit AttributeError.  This is really a custom error based on KB logic, just create a custom exception IMO.  And don't prefix the name with "KegBouncer", it's already namespaced by python modules.  `PrimaryKeyError` seems sufficient IMO.  But, you may want that to inherit from a generic `KegBouncerException` so that library users can easily catch any exception thrown from the library.